pennylane/ops/qubit/matrix_ops.py

```diff
@@ -332,3 +334,153 @@ def _controlled(self, control):
 
     def label(self, decimals=None, base_label=None, cache=None):
         return super().label(decimals=decimals, base_label=base_label or "U", cache=cache)
+
+
+class BlockEncode(Operation):
+    r"""BlockEncode(a, wires)
+    Apply an arbitrary matrix, :math:`A`, encoded in the top left block of a unitary matrix.
+
+    .. math::
+
+        \begin{align}
+             U(A) &=
+             \begin{bmatrix}
+                A & \sqrt{I-AA^\dagger} \\
+                \sqrt{I-A^\dagger A} & -A^\dagger
+            \end{bmatrix}.
+        \end{align}
```

Comment:
I think we include a sentence about the mathematical properties of `A`, being an `(n, m)` matrix with complex coefficients, just to be explicit, and maybe mention `ceil(log2(n+m))` qubits as well! Not a blocker, though.